commcare-cloud/commcare_cloud/environment/schemas/postgresql.py

```diff
@@ -117,5 +122,34 @@ class SynclogsDBOptions(DBOptions):
     django_alias = 'synclogs'
 
 
-class ShardDBOptions(DBOptions):
+class FormProcessingConfig(jsonobject.JsonObject):
+    _allow_dynamic_properties = False
+    proxy = jsonobject.ObjectProperty(lambda: FormProcessingProxyDBOptions, required=True)
+    partitions = jsonobject.DictProperty(lambda: StrictPartitionDBOptions, required=True)
+
+    @classmethod
+    def wrap(cls, data):
+        for i, (django_alias, db) in enumerate(data['partitions'].items()):
+            db['django_alias'] = db.get('django_alias', django_alias)
+            db['name'] = db.get('name', 'commcarehq_{}'.format(db['django_alias']))
+        self = super(FormProcessingConfig, cls).wrap(data)
+        return self
+
+    def get_db_list(self):
+        return [self.proxy] + self.partitions.values()
+
+
+class FormProcessingProxyDBOptions(DBOptions):
+    name = constants.form_processing_proxy_db_name
+    django_alias = 'proxy'
+
+
+class PartitionDBOptions(DBOptions):
     shards = jsonobject.ListProperty(int, exclude_if_none=True)  # [start, end] pair
+
+
+class StrictPartitionDBOptions(PartitionDBOptions):
+    def validate(self, *args, **kwargs):
+        assert re.match(r'p\d+', self.django_alias)
```

Comment:
should this say `re.match(r'p\d+$'`? as is it will match values like `p1_hey_there`

Reply:
Yes I did mean that. I thought match implied `^` and `$`, but it looks like it just implies `^`. Go figure. Thanks!